include/solvers/vof.h

```diff
@@ -369,10 +378,40 @@ class VOF : public AuxiliaryPhysics<dim, TrilinosWrappers::MPI::Vector>
   virtual void
   copy_local_rhs_to_global_rhs(const StabilizedMethodsCopyData &copy_data);
 
+  /**
+   * @brief Limit the phase fractions between 0 and 1. This is necessary before interface sharpening
+   */
+  void
+  update_solution_and_constraints(TrilinosWrappers::MPI::Vector &solution);
+
+  /**
+   * @brief Assemble the system for interface sharpening
+   *  * This function assembles the weak form of:
+   *  $$ \Phi = c ^ (1 - \alpha) * (\phi ^ \alpha)  if 0 <=
+   * \phi <= c $$
+   *  $$ \Phi = 1 - (1 - c) ^ (1 - \alpha) * (1 - \phi) ^ \alpha  if c <
+   * \phi <= 1 $$
+   * Reference for sharpening method
+   * https://www.sciencedirect.com/science/article/pii/S0045782500002000
+   */
+  void
+  assemble_L2_projection_interface_sharpening(
+    TrilinosWrappers::MPI::Vector &solution);
+
+  /**
+   * @brief Solve the assembled system to sharpen the interface
+   */
+  void
+  solve_interface_sharpening(TrilinosWrappers::MPI::Vector &solution);
+
```

Comment:
Add a comment here to explain what this function does.

Reply:
added.